evalml/pipelines/components/transformers/preprocessing/replace_nullable_types.py

```diff
@@ -97,9 +97,4 @@ def fit_transform(self, X, y=None):
         Returns:
             tuple of pd.DataFrame, pd.Series: The input features and target data with the non-nullable types set.
         """
-        X_ww = infer_feature_types(X)
-        if y is not None:
-            y_ww = infer_feature_types(y)
-        else:
-            y_ww = y
-        return self.fit(X_ww, y_ww).transform(X_ww, y_ww)
+        return self.fit(X, y).transform(X, y)
```

Comment:
Both `fit` and `transform` already have `infer_feature_types` calls, these felt redundant